packages/server/graphql/types/User.ts

```diff
@@ -421,19 +423,57 @@ const User = new GraphQLObjectType<any, GQLContext>({
       },
       resolve: async ({id: userId}, {reflectionId}, {dataLoader}) => {
         const retroReflection = await dataLoader.get('retroReflections').load(reflectionId)
-        if (!retroReflection) return []
+        if (!retroReflection) {
+          return standardError(new Error('Invalid reflection id'), {userId})
+        }
         const {meetingId} = retroReflection
         const meetingMemberId = toTeamMemberId(meetingId, userId)
-        const [viewerMeetingMember, reflectionGroups] = await Promise.all([
+        const r = await getRethink()
+        const [viewerMeetingMember, reflections] = await Promise.all([
           dataLoader.get('meetingMembers').load(meetingMemberId),
-          dataLoader.get('retroReflectionGroupsByMeetingId').load(meetingId)
+          r
+            .table('RetroReflection')
+            .getAll(meetingId, {index: 'meetingId'})
+            .filter({isActive: true})
+            .orderBy('createdAt')
+            .run()
         ])
         const {meetingId: viewerMeetingId} = viewerMeetingMember
         if (viewerMeetingId !== meetingId) {
```

Comment:
The `meetingId` should always be the same but the `viewerMeetingMember` uses the userId that we get from the source. If the viewer passes a `reflectionId` to the query that lives in a meeting that they're not a part of, the viewer's meeting id should be different.

Now that I'm looking at it, I don't like using `toTeamMemberId` for meeting members so I've added a new helper function

Reply:
> If the viewer passes a reflectionId to the query that lives in a meeting that they're not a part of, the viewer's meeting id should be different.

How do we test this case in the code?

We deconstruct the `meetingId` from the reflection in line 430 and then use that to construct a `meetingMemberId` in line 431. If the user is **not** in that meeting, then line 434 should return a null object, which will cause line 442 blow up, yes? On the other hand, if the use is indeed in that meeting, the returned `viewerMeetingMember` object will always have the same `meetingId` as the one we deconstructed from line 430, so the check in line 443 is not helpful.
Tell me if I go completely nut and become blinded in this part of code...

Reply:
True @tianrunhe! I've updated the check so that the error is thrown if `viewerMeetingMember` is null. I tested this by changing `reflectionId` to an id that belongs in a meeting that the viewer is not a part of. I'll move this to Maintainer review unless there's you have any other feedback?